app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,113 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy approve disapprove]
+  before_action :authenticate_user!
+  # GET /posts
+  # GET /posts.json
+  def index
+    if current_user.is_admin
+      @posts = Post.all
+    else
+      @posts = current_user.posts
+    end
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
+  end
+
+  # GET /posts/new
+  def new
+    @categories = Category.all
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit
+    @categories = Category.all
+  end
+
+  # POST /posts
+  # POST /posts.json
+  def create
+    #validate ajax if images nil?
+    if post_params[:images].blank?
+      @errors = "Images source not blank.Please choose image before update"
+      return
+    end
+
+    @url_image = convert_to_url_cloudinary(post_params[:images].original_filename)
+
+    @category = Category.find(params[:category_id])
+
+    @post = @category.posts.new(post_params.merge(images: @url_image, user_id: current_user.id))
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to @post, notice: 'Post was successfully created.' }
+        format.json { render :show, status: :created, location: @post }
+      else
+        format.html { render :new }
+        format.json { render json: @post.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /posts/1
+  # PATCH/PUT /posts/1.json
+  def update
+    @url_image = nil
+    if post_params[:images].blank?
+      @url_image = Post.find(params[:id])[:images]
+    else
+      @url_image = convert_to_url_cloudinary(post_params[:images].original_filename)
+    end
```

Comment:
Please update your code with this convention: https://github.com/rubocop-hq/ruby-style-guide#returning-result-from-ifcase